Add -no-color flag

jabrown85 opened the pull request:
Issue: #312

I wasn't entirely sure how best to test this. I am happy to add one anywhere someone suggests tho!

Signed-off-by: Jesse Brown <jabrown85@gmail.com>

jabrown85 commented:
I realize `pack` also strips colored output..is that something we want lifecycle to do as well? This PR gets rid of all the colored output that lifecycle emits, but it doesn't do anything with the output by the buildpacks themselves.

ekcasey commented:
I don't think `pack` _successfully_ strips color?
![Screen Shot 2020-07-02 at 12 58 44 PM](https://user-images.githubusercontent.com/1994518/86389626-96bde400-bc64-11ea-9090-119c49bc91a2.png)

ekcasey commented on cmd/command.go:
```diff
@@ -17,11 +19,14 @@ func Run(c Command, asSubcommand bool) {
 	FlagVersion(&printVersion)
 	FlagLogLevel(&logLevel)
+	FlagNoColor(&noColor)
+	color.Disable(noColor)
```
This need to happen after `flagSet.Parse`, `noColor` won't be set correctly at this point in the command execution.

jabrown85 replied:
ah, I moved it right after my last smoke test. Good catch.

ekcasey commented on cmd/flags.go:
```diff
@@ -156,6 +156,10 @@ func FlagProcessType(processType *string) {
 }
 
+func FlagNoColor(skip *bool) {
+	flagSet.BoolVar(skip, "no-color", false, "disable color output")
```
We should add an environment variable `CNB_NO_COLOR` and use it to set the default here. This is especially for things that might not be specified in the platform API, as it allows platforms to optimistically set it w/o worrying that it will cause a failure if it is unsupported by a given lifecycle.

jabrown85 commented:
Added env support and fixed the ordering. Things look good as far as lifecycle emitted colors go.

![image](https://user-images.githubusercontent.com/790999/86390886-7a1ead80-bc5e-11ea-87dd-28975e78b433.png)

I think we should handle stripping colors later, if it is even reasonable to do.

jromero commented:
> I don't think `pack` _successfully_ strips color?
> ![Screen Shot 2020-07-02 at 12 58 44 PM](https://user-images.githubusercontent.com/1994518/86389626-96bde400-bc64-11ea-9090-119c49bc91a2.png)
> 
> If we could strip the buildpack color that would be a big win. Folks have been complaining about this for some time ([buildpacks/pack#123](https://github.com/buildpacks/pack/issues/123)) and while `pack` may have (temporarily?) fixed it, this should be supported in all platforms.

`pack` did recently ["fix"](https://github.com/buildpacks/pack/pull/696) this, to be released in 0.12.0, but having the option and functionality in lifecycle would be a huge benefit.

jromero commented:
@ekcasey out of curiosity is this a "platform API" change? I would be curious to know, as a platform that could support other lifecycle implementations, how we would know if this option is available if not in the API.

ekcasey commented:
@jromero that's a good question. I don't think we want to specify this flag I don't think a compliant lifecycle should have to have color.

I want to stake out a new philosophical principal. When using unspecified features a platform _should_ use the environment variables rather than the flags to provide inputs. There is no harm in setting an env var if the lifecycle doesn't support the feature. `pack` can just optimistically provide the env var.
